MyPerfectOnboardingApplication/Sources/MyPerfectOnboarding.Api/DependencyResolver.cs

```
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Web.Http.Dependencies;
+using Unity;
+using Unity.Exceptions;
+
+namespace MyPerfectOnboarding.Api
+{
+    public class DependencyResolver : IDependencyResolver
+    {
+        protected IUnityContainer _container;
+
+        public DependencyResolver(IUnityContainer container)
+        {
+            _container = container;
+        }
+
+        public IDependencyScope BeginScope()
+        {
+            var child = _container.CreateChildContainer();
+            return new DependencyResolver(child);
+        }
+
+        public void Dispose()
+        {
+            _container.Dispose();
+        }
+
+        public object GetService(Type serviceType)
+        {
+            try
+            {
+                return _container.Resolve(serviceType);
+            }
+            catch (ResolutionFailedException)
```

Please, have a short debug session with this catch - I know you have a test for registered dependencies, but you should still let your code [fail fast](https://dzone.com/articles/fail-fast-principle-in-software-development) in case some dependency is missing. By returning `null` (or empty collection below) you actually mask where an error originates and mask root cause. You will find that there is a reason why the boilerplate looks like this, however, it would be much better to throw `ResolutionFailedException` when a type is forgotten to be registered…

I will insist on this in the next task, consider postponing it until you separate `UnityContainer`…

---

https://stackoverflow.com/questions/3819842/catch-exception-with-condition